### PR TITLE
Be more deterministic in producing spell checking results.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This should hopefully address a problem where we get different test results on 64bit vs 32bit.

### CODE REVIEW

File: src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs [hunk 1]
```diff
@@ -272,7 +272,7 @@ public static IEnumerable<T> Flatten<T>(this IEnumerable<IEnumerable<T>> sequenc
 
         public static IEnumerable<T> OrderBy<T>(this IEnumerable<T> source, IComparer<T> comparer)
         {
-            return source.OrderBy(t => t, comparer);
+            return source.OrderBy(Functions<T>.Identity, comparer);
```
Comment (agocke) [inline src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs]: The BCL seriously still has no identity function?

**Comment by agocke** [comment]
The compiler-area changes :+1: .

I'm clueless when it comes to Features, but IMHO I'm not convinced this change fixes the presumptive problem.

I'd like to see at least one test case that fails without the change.

**Comment by Pilchie** [comment]
Agree with Andy.  Why do you think this will be more stable?

**Comment by brettfo** [comment]
@agocke: The test [`SpellcheckTests.SuggestEscapedPredefinedTypes()`](http://source.roslyn.io/#Roslyn.Services.Editor.VisualBasic.UnitTests/Diagnostics/Spellcheck/SpellcheckTests.vb,c9ea866f17d7ac39) will regularly fail on x64 without this PR and will pass with it applied.

@Pilchie: As far as I can tell the instability is ultimately due to the order that [`ImmutableHashSet`'s](https://msdn.microsoft.com/en-us/library/dn467171(v=vs.111).aspx) enumerator returns the items (as exposed through [`MultiDictionary<K, V>`](http://source.roslyn.io/#Microsoft.CodeAnalysis/InternalUtilities/MultiDictionary.cs,33aeb68ec0af145a)), but as to how this PR actually fixes that problem is beyond me.  It appears that these changes simply extract some lambdas to static functions with the same body.

FYI @tannergooding.

**Comment by agocke** [comment]
@brettfo  [`SpellcheckTests.SuggestEscapedPredefinedTypes()`](http://source.roslyn.io/#Roslyn.Services.Editor.VisualBasic.UnitTests/Diagnostics/Spellcheck/SpellcheckTests.vb,c9ea866f17d7ac39) fails on x64 without this patch *now*, but without a mechanism of action, I see no reason why we couldn't silently break this functionality in the future.

**Comment by CyrusNajmabadi** [comment]
@brettfo @Pilchie This is more stable due to the ```.Order``` addition here:

```c#
             var matches = results.OrderBy(kvp => kvp.Key)
-                                 .SelectMany(kvp => kvp.Value)
+                                 .SelectMany(kvp => kvp.Value.Order())
                                  .Where(t => t != nameText)
                                  .Take(3)
                                  .Select(n => CreateCodeAction(nameNode, nameText, n, document));
```

Here's what was happening before: 

We were getting results ordered by their weight.  But for any results *with the same weight* the order of results was non-deterministic.  Now elements with the same weight are also ordered.

**Comment by Pilchie** [comment]
Ah - so easy to miss that in the diff :Smile:

:shipit:

**Comment by brettfo** [comment]
@CyrusNajmabadi can you unskip this test: [`SpellcheckTests.SuggestEscapedPredefinedTypes()`](http://source.roslyn.io/#Roslyn.Services.Editor.VisualBasic.UnitTests/Diagnostics/Spellcheck/SpellcheckTests.vb,c9ea866f17d7ac39) now that the determinism is fixed?

**Pull request merged**